server/src/main/java/com/vaadin/data/Binder.java

```diff
@@ -1285,7 +1285,10 @@ private void handleFieldValueChange(
                 result.ifOk(value -> {
                     setter.accept(bean, value);
                     if (value != null) {
-                        getField().setValue(convertToFieldType(value));
+                        FIELDVALUE converted = convertToFieldType(value);
+                        if (!field.getValue().equals(converted)) {
```

Comment:
hello! this instruction raises NullPointerException when field.getValue() is null. should i open a new issue for it? thank you!

Reply:
This is fixed in 8.12.4

Reply:
thank you!